### ISSUE TITLE
PHPUnit 6 upgrade fails 5.7 passing tests

### ISSUE THREAD

**Issue by guilhermeblanco** [sebastianbergmann/phpunit#2482]
As soon as I attempt to upgrade to PHPUnit 6 (and quickly do necessary changes to make it run), the following test starts to fail in Doctrine ORM: https://github.com/doctrine/doctrine2/blob/develop/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php

Here's the output:

```bash
[gblanco@VM5801 ~/workspaces/doctrine/orm (develop *)]$ ./vendor/bin/phpunit tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php
PHPUnit 6.1-g28c5c57db by Sebastian Bergmann and contributors.

.....................FF.FFFF..............F...                    46 / 46 (100%)

Time: 91 ms, Memory: 6.00MB

There were 7 failures:

1) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #21 ('clear', array())
Failed asserting that null matches expected 'INNER VALUE FROM clear'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

2) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #22 ('clear', array('all'))
Failed asserting that null matches expected 'INNER VALUE FROM clear'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

3) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #24 ('persist', array('req'))
Failed asserting that null matches expected 'INNER VALUE FROM persist'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

4) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #25 ('remove', array('req'))
Failed asserting that null matches expected 'INNER VALUE FROM remove'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

5) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #26 ('refresh', array('req'))
Failed asserting that null matches expected 'INNER VALUE FROM refresh'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

6) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #27 ('detach', array('req'))
Failed asserting that null matches expected 'INNER VALUE FROM detach'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

7) Doctrine\Tests\ORM\Decorator\EntityManagerDecoratorTest::testAllMethodCallsAreDelegatedToTheWrappedInstance with data set #42 ('initializeObject', array('req'))
Failed asserting that null matches expected 'INNER VALUE FROM initializeObject'.

/Users/gblanco/workspaces/doctrine/orm/tests/Doctrine/Tests/ORM/Decorator/EntityManagerDecoratorTest.php:73

FAILURES!
Tests: 46, Assertions: 85, Failures: 7.
```

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by guilhermeblanco** [comment]
@sebastianbergmann This issue can be closed, however I'm adding the reasoning of the issue here. I only found it after trying to reproduce in a minimal, self-contained test case:

EntityManagerDecorator is an abstract class that delegates all calls to EntityManagerInterface. Some methods (such as clear(), persist()) do not return the output from the wrapped EntityManagerInterface instance. This leads to pass in PHPUnit 5.7 but fail on PHPUnit 6.0 (basically,  version 6 addressed a bug from 5.7).
